### ISSUE TITLE
Incorrect port when BASE_URL contains protocol's default port

### ISSUE THREAD

**Issue by pedorich-n** [FoxxMD/multi-scrobbler#155]
**Describe the bug**
I am running the `multi-scrobbler` behind the reverse proxy on the address `http://multi-scrobbler.server.lan` and specify the same address in `BASE_URL` (also tried `http://multi-scrobbler.server.lan:80`).

Two issues arise:
1. Because [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL/port) trims the default protocol's port (80 for HTTP and 443 for HTTPS), MS assumes I didn't specify the port and tries to use its default port for the base url instead. Which is not correct in my case: https://github.com/FoxxMD/multi-scrobbler/blob/0dcd049453fcbb0e8cadfe8978f5d822b069ce1d/src/backend/ioc.ts#L74-L79 
3. The `items.mainPort` in the code mentioned above doesn't exist, so I ended up with `http://multi-scrobbler.server.lan:undefined` as a base url for redirects. Looks like [this commit](https://github.com/FoxxMD/multi-scrobbler/commit/c98672bcef5414e6383acff0517d9060b94b8329#diff-02f46a82b21d52379bd82e55a585501c40b3c7293798cbb1e12fff4d7db4457bR61) caused this issue.

**Expected behavior**
Use the default port when specified.

**Logs**
```
[App] [API] User-defined base URL for UI and redirect URLs (spotify, deezer, lastfm): http://multi-scrobbler.server.lan:undefined
```

**Versions (please complete the following information):**

- multi-scrobbler: 0.7.1, docker

**Additional context**
The port gets lost after [normalization](https://github.com/FoxxMD/multi-scrobbler/blob/0dcd049453fcbb0e8cadfe8978f5d822b069ce1d/src/backend/ioc.ts#L74) because the `normalize-url` [also uses `URL`](https://github.com/sindresorhus/normalize-url/blob/4869023f4f2766b58e1954fd30e5f1609d2cf8e7/index.js#L119).

I think the code should either assume the port is 80 when not specified, as it seems to be the behavior of `URL`, or at least respect the explicitly specified port even when it's the default.

The current workaround for me is to explicitly specify the redirect URLs for sources.

**Comment by FoxxMD** [comment]
Thanks for catching this and the detailed write up!

**Comment by FoxxMD** [comment]
@pedorich-n please try `foxxmd/multi-scrobbler:develop` docker image and let me know if that fixes URL issues for you

**Comment by pedorich-n** [comment]
Thanks for such a quick fix!
The URL does appear to be correct now, with the `BASE_URL=http://multi-scrobbler.server.lan:80`:
```
[App] [API] User-defined base URL for UI and redirect URLs (spotify, deezer, lastfm): http://multi-scrobbler.server.lan/
...
[App] [Sources] [Spotify - spotify] Redirect URL that will be used on auth callback: 'http://multi-scrobbler.server.lan/callback'
```


But I am getting a 400 HTTP code from MS when trying to authorize Spotify now. I don't know if that's something on MS's side or my reverse proxy or what... The URL I see is `http://multi-scrobbler.server.lan/api/callback?code=<code>&state=spotify`. Where did the `/api/` come from? 🤔 

In the logs, I can see
```
[2024-06-05 20:07:04.265 +0900] INFO   : [App] [API] Received auth code callback from Spotify
[2024-06-05 20:07:02.364 +0900] INFO   : [App] [API] Redirecting to spotify authorization URL
```
But the page still shows "Auth Interaction Required" and no `currentCreds-spotify.json` file is created.

**Comment by FoxxMD** [comment]
Oops! Missed a string conversion of the base url when passing to the spotify api. Try the newest `develop` tag, it should be fixed now.

**Comment by pedorich-n** [comment]
Yes, it works now!
Thank you for the fix!

**Comment by FoxxMD** [comment]
Released in [v0.8.0](https://github.com/FoxxMD/multi-scrobbler/releases/tag/0.8.0)
